Introduction + Getting started (#25)
primer/contribute: content/index.mdx

---
title: Getting started
description: Contributions to Primer can take many different formats to align with your skills and available time.
---

import {HeroLayout} from '@primer/gatsby-theme-doctocat'

export default HeroLayout

There are several ways you can contribute to and participate in Primer:

1. Participate in discussions
2. Propose a new UI pattern
3. Improve the documentation
4. Design or build new patterns
5. Give us feedback and report bugs 
6. Contribute to our open source repos

All contributions to Primer need to follow our [code of conduct](https://github.com/github/primer/blob/main/CODE_OF_CONDUCT.md). 

If at any time you get stuck or need help, head to #primer on Slack or start a discussion in github/primer with your question.

<Note>Please note that at this time, we are not looking for external contributions from non-GitHub staff. Some of the links included in these documents are only available to GitHub staff.</Note>

## Participate in discussions

If you have a question, feedback or suggestion, you can start a new [discussion](https://github.com/github/primer/discussions) (GitHub staff only), or have a look at existing ones.

> The Design Infrastructure team holds office hours once a week and anyone at GitHub can join, ask questions and participate in ongoing conversations. The office hours Zoom invite links are posted in Slack before the meeting starts.

Reviewer: Given it's a public repo, not sure if we should reference internal team names. Shall we change this to Primer? cc. @lesliecdubs.

Author: Good point, we reference to the Design Infrastructure team as well in the [Documentation voice and tone](https://primer.style/contribute/documentation#:~:text=Use%20%22we%22%20consistently%20to%20refer%20to%20the%20Design%20Systems%20/%20Infrastructure%20team%2C%20not%20to%20GitHub%20as%20a%20whole.)

Reviewer 2: We also reference internal team names on the [About page](https://primer.style/about/#team/), but that seems more natural since we're specifically talking about the team.

I'd tend to agree with going more generic when we can. Suggestion: "Primer office hours are held once a week..."

## Propose a new UI pattern

If you'd like to propose a new Primer pattern, big or small, let's talk! The best way to get started, especially if your proposal is in its early stages, is to start a new [discussion](https://github.com/github/primer/discussions) (GitHub staff only).

If you're more certain about what you need, please open a [pattern request issue](https://github.com/github/primer/issues/new?assignees=&labels=type%3A+request&template=0-request.md&title=%5BRequest%5D+) (GitHub staff only). We will get back to you after our weekly backlog triaging session.

## Improve the documentation

Documentation is a core part of Primer, and, just as design and code, we’re always trying to make it better and more useful. If you notice something missing, a typo, or have an idea of how the docs can be improved, please start a new [discussion](https://github.com/github/primer/discussions), or submit a pull request with a fix (you can do this directly via the “Edit this page on GitHub” link on the footer of docs pages).

Read the Primer guidelines for [writing documentation](https://primer.style/contribute/documentation).

## Design or build new patterns

> You can contribute to new patterns either by design, prototype and build proofs of concept, test in dotcom, and contribute directly to our Primer open source repos.

Reviewer: Don't think we should refer to 'dotcom' in public repo's, can we remove that bit? 

Also on this line, was struggling to scan it. Is it worth keeping it shorter and non-specific?

```suggestion
You can contribute to our Primer open source repos in many ways.
```

Author: - True, I'll remove the `doctom` part. 
- On keeping the other paragraph shorter, I like the idea of listing the different ways you can contribute rather than keeping the text short, but it's true it looks a bit verbose. What about:

`You can contribute to new patterns either by design, prototype and build proofs of concept to our Primer open source repos`
